packages/transaction-controller/src/utils.ts

```diff
@@ -209,26 +209,19 @@ export function getAndFormatTransactionsForNonceTracker(
 }
 
 /**
- * Checks whether a given transaction matches the specified network or chain ID.
- * This function is used to determine if a transaction is relevant to the current network or chain.
+ * Checks whether a given transaction matches the specified chain ID.
+ * This function is used to determine if a transaction is relevant to the current chain.
  *
  * @param transaction - The transaction metadata to check.
  * @param chainId - The chain ID of the current network.
- * @param networkId - The network ID of the current network.
- * @returns A boolean value indicating whether the transaction matches the current network or chain ID.
+ * @returns A boolean value indicating whether the transaction matches the current chain ID.
  */
-export function transactionMatchesNetwork(
+export function transactionMatchesChainId(
   transaction: TransactionMeta,
   chainId: Hex,
-  networkId: string | null,
 ) {
-  if (transaction.chainId) {
-    return transaction.chainId === chainId;
-  }
-  if (transaction.networkID) {
-    return transaction.networkID === networkId;
-  }
-  return false;
+  const txChainId = transaction.chainId ?? transaction.txParams.chainId;
```

Comment:
Also here, I worry this callback just adds additional complexity as the `chainId` in both locations is set in code using the network state.

Reply:
this just becomes
```
export function transactionMatchesChain(
  transaction: TransactionMeta,
  chainId: Hex,
) {
  return chainId === transaction.chainId;
}
```

no need for this function at that point?

Reply:
getting rid of this helper since it's no longer really necessary